Remotion/Web/Development.WebTesting/Accessibility/AccessibilityAnalyzer.cs

```diff
@@ -241,6 +243,35 @@ private bool IsEmptyFrame ()
       return !WebDriver.FindElements (By.XPath ("/html/body/*")).Any();
     }
 
+    private void WaitUntilFrameIsVisible (int timeout = 3000)
+    {
+      var webDriverWait = new WebDriverWait (WebDriver, TimeSpan.FromMilliseconds(timeout))
+                          {
+                              PollingInterval = TimeSpan.FromMilliseconds (timeout / 10)
+                          };
+
+      try
+      {
+        webDriverWait.Until (FrameLoaded);
+      }
+      finally
+      {
+        WebDriver.SwitchTo().DefaultContent();
+      }
+    }
+
+    /// <summary>
+    /// Sometimes, due to the delayed generation of the compiled aspx file, the content of the DatePicker iframe is not fully loaded when the iframe is rendered.
+    /// </summary>
+    private bool FrameLoaded (IWebDriver driver)
+    {
+      driver.SwitchTo().DefaultContent();
+      var iframe = driver.FindElement(By.TagName("iframe"));
```

Comment:
This workaround only ensures that the *first* `iframe` is loaded. Imo `WaitUntilFrameIsVisible` should accept the frame element as parameter instead of querying anew (if possible).

Reply:
I was thinking it was fine because I am querying from within the loop (L231) and somehow thought that restricted to scope to the "current" `iframe`. But what you say makes sense - I am querying for the _first_ `iframe` over and over again, correct? In that case I will give it a try to wait based on the each "current" `iframe` of the loop.